org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/hcr/ScalaHotCodeReplaceManager.scala

```diff
@@ -150,8 +152,17 @@ private[internal] trait HotCodeReplaceExecutor extends Publisher[HCRResult] with
     }
   }
 
+  protected val doHotCodeReplaceBarrier = new AtomicBoolean
+  private def waitForBarrier(): Unit = {
+    if (doHotCodeReplaceBarrier.getAndSet(true)) {
+      while (doHotCodeReplaceBarrier.get) {}
```

Comment:
actually, it is very rare situation and I think related to test rather not real user work. I do not have any statistics.